lib/ebml/tools.js

```diff
@@ -89,35 +91,33 @@ module.exports = class Tools {
         const result = new Uint8Array(a1.byteLength + a2.byteLength);
         result.set(a1, 0);
         result.set(a2, a1.byteLength);
+        // xxx: Why are these nulled out?
```

Comment:
I really have no idea. This should not do anything. 
It came from here: https://github.com/jayands/node-ebml/commit/cf9ecb1b8db51af4c1cd084367b15dd2b0a9c07f#diff-a03fbe7ce44319924b46e517060f752aR78
Maybe @OllieJones knows something we do not?

Reply:
You're not entirely wrong; I looked up how JS handles memory and it is *super* weird. Primitives are passed by value, but complex objects and arrays as a kind of ["copy of a reference"](https://stackoverflow.com/a/13104500). But because of that, it's possible  to modify the variable and have the mutation stick.